shop/src/main/java/ro/msg/learning/shop/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.entity.Product;
+
+import java.util.*;
+@Service
+public class ProductServiceImpl implements ProductService {
+    private static final Map<UUID, Product> productRepository = new HashMap<>();
+
+    @Override
+    public void createProduct(Product product) {
+        productRepository.put(product.getId(), product);
+    }
+
+    @Override
+    public void updateProduct(UUID id, Product product) {
```

Comment:
In this documentation: https://spring.io/guides/tutorials/rest/ you will find good examples to implement these methods, example: replaceEmployee for update